presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -290,6 +290,11 @@ public void addTypes(IdentityHashMap<Expression, Type> types)
         this.types.putAll(types);
     }
 
+    public void replaceType(Expression expression, Type type)
```

Comment:
I thought so too.
But if I remove this line, the non-equi join criteria coercions will not work, and throw exception.

In function appendProjections in RelantionPlanner, 
Symbol symbol = symbolAllocator.newSymbol(expression, analysis.getType(expression));
If I dont's explicity replace the types map, analysis.getType(expression) will still get the origin type of expression but not the coercion type.

Any idea about this?